packages/contracts/contracts/voting/majority/MajorityVotingBase.sol

```diff
@@ -308,23 +303,26 @@ abstract contract MajorityVotingBase is
         if (_isVoteOpen(proposal_)) {
             // Early execution
             return
-                worstCaseSupport(_proposalId) > proposal_.supportThreshold &&
-                participation(_proposalId) >= proposal_.minParticipation;
+                proposal_.configuration.voteMode == VoteMode.EarlyExecution &&
+                worstCaseSupport(_proposalId) > proposal_.configuration.supportThreshold &&
+                participation(_proposalId) >= proposal_.configuration.minParticipation;
         } else {
             // Normal execution
             return
-                support(_proposalId) > proposal_.supportThreshold &&
-                participation(_proposalId) >= proposal_.minParticipation;
+                support(_proposalId) > proposal_.configuration.supportThreshold &&
+                participation(_proposalId) >= proposal_.configuration.minParticipation;
         }
     }
 
     /// @notice Internal function to check if a proposal vote is still open.
     /// @param proposal_ The proposal struct.
     /// @return True if the proposal vote is open, false otherwise.
     function _isVoteOpen(Proposal storage proposal_) internal view virtual returns (bool) {
+        uint64 currentTime = getTimestamp64();
+
         return
-            getTimestamp64() < proposal_.endDate &&
-            getTimestamp64() >= proposal_.startDate &&
+            proposal_.configuration.startDate <= currentTime &&
```

Comment:
`proposal_.configuration.startDate < currentTime + 1` would be gas wise cheaper than `proposal_.configuration.startDate <= currentTime`

Reply:
After discussion with @novaknole, we won't do such optimizations now.